pkg/api/client/client.go

```diff
@@ -96,10 +104,30 @@ func NewDefault(addr string, opts ...Option) *DefaultClient {
 	for _, opt := range opts {
 		opt(c)
 	}
+
+	t := http.DefaultTransport
+
+	// change transport to dial to the unix socket instead
+	unixSocketFile := api.ExtractUnixSocket(addr)
+	unixIdent := "unix"
+	if unixSocketFile != "" {
+		t = &http.Transport{
+			DialContext: func(ctx context.Context, _, _ string) (net.Conn, error) {
+				var d net.Dialer
+				return d.DialContext(ctx, unixIdent, unixSocketFile)
+			},
+		}
+		// also make sure to unset the address and modify the scheme so the calls to NewURL
+		// don't append the filename of the unix socket
+		c.hostAddr = ""
+		c.scheme = c.scheme + unixIdent
```

Comment:
Maybe also use something like `filepath.Join()` or `path.Join` to construct safe(r) URLs?

Reply:
Did that at first. What happened was that it turned `http://` into `http:/` 😦